### PR TITLE
Evikulov/readme update

### PR DESCRIPTION
**Pull request by EVikulov**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
   - Call `php artisan swagger:push-documentation` console command after the `tests` stage in your CI/CD configuration
 
 ## Usages
- For correct working of plugin you have to dispose all the validation rules in the rules() method of `YourRequest` class, 
- which must be connected to the controller via DependencyInjection. In annotation of custom request you can specify 
- summary and description. Plugin will take validation rules from your request and use it as description 
- of input parameter. 
+To collect documentation run following steps:
   
 ### Example
```
Comment (astorozhevsky) [inline readme.md:55]: please remove this heading

File: readme.md [hunk 2]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
   - Call `php artisan swagger:push-documentation` console command after the `tests` stage in your CI/CD configuration
 
 ## Usages
```
Comment (astorozhevsky) [inline readme.md:52]:
```suggestion
## Usage
```

File: readme.md [hunk 3]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
- For correct working of plugin you have to dispose all the validation rules in the rules() method of `YourRequest` class, 
- which must be connected to the controller via DependencyInjection. In annotation of custom request you can specify 
- summary and description. Plugin will take validation rules from your request and use it as description 
- of input parameter. 
+To collect documentation run following steps:
```
Comment (astorozhevsky) [inline readme.md:53]:
```suggestion
```

File: readme.md [hunk 4]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
   
 ### Example
+1. Create request data fixture
+ ```
```
Comment (astorozhevsky) [inline readme.md:57]:
```suggestion
 ```json
```

File: readme.md [hunk 5]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+ {
+    "first_name": "Updated User",
+    "is_active": true,
+    "age": 22
+ }
+ ```
```
Comment (astorozhevsky) [inline readme.md:63]:
```suggestion
 ```

```

File: readme.md [hunk 6]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+2. Prepare test case code of API endpoint
```
Comment (astorozhevsky) [inline readme.md:64]:
```suggestion
2. Create test for testing API endpoint:
```

File: readme.md [hunk 7]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+ ```php
+    public function testUpdate()
```
Comment (astorozhevsky) [inline readme.md:66]:
```suggestion
public function testUpdate()
```

File: readme.md [hunk 8]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+    {
```
Comment (astorozhevsky) [inline readme.md:67]:
```suggestion
{
```

File: readme.md [hunk 9]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+        $data = json_decode(file_get_contents('update_user.json'), true);
```
Comment (astorozhevsky) [inline readme.md:68]:
```suggestion
    $data = json_decode(file_get_contents('update_user.json'), true);
```

File: readme.md [hunk 10]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+
+        $response = $this->json('put', '/users/1', $data);
```
Comment (astorozhevsky) [inline readme.md:70]:
```suggestion
    $response = $this->json('put', '/users/1', $data);
```

File: readme.md [hunk 11]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
 
+        $response->assertStatus(Response::HTTP_NO_CONTENT);
```
Comment (astorozhevsky) [inline readme.md:72]:
```suggestion
    $response->assertStatus(Response::HTTP_NO_CONTENT);
```

File: readme.md [hunk 12]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+    }
```
Comment (astorozhevsky) [inline readme.md:73]:
```suggestion
}
```

File: readme.md [hunk 13]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+ ```
```
Comment (astorozhevsky) [inline readme.md:74]:
```suggestion
 ```

```

File: readme.md [hunk 14]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+3. Create request file code.
+
+For correct working of plugin you have to dispose all the validation rules in the rules() method of `YourRequest` class,
```
Comment (astorozhevsky) [inline readme.md:77]:
```suggestion
For correct working of plugin you have to dispose all the validation rules in the `rules()` method of `UpdateUserDataRequest` class,
```

File: readme.md [hunk 15]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+which must be connected to the controller via DependencyInjection. In annotation of custom request you can specify
```
Comment (astorozhevsky) [inline readme.md:78]:
```suggestion
which must be connected to the controller via Dependency Injection.
```

File: readme.md [hunk 16]
```diff
@@ -124,7 +145,12 @@ plugin is able to draw Swagger-template to display the generated documentation f
  - *auto-doc.route* - it's a route for generated documentation  
  - *auto-doc.basePath* - it's a root of your api root
  
-Also you can specify way to collect documentation by creating your custom data collector class.
+Also, you can specify way to collect documentation by creating your custom data collector class.
+
```
Comment (astorozhevsky) [inline readme.md:149]: let's add example of controller method:

```php
public function update(UpdateUserDataRequest $request)
{
     ...
}
```

File: readme.md [hunk 17]
```diff
@@ -107,7 +128,7 @@ plugin is able to draw Swagger-template to display the generated documentation f
  - **@summary** - short description of request
```
Comment (astorozhevsky) [inline readme.md:128]: need to add the sentence before:

```md
You can use the following annotations in your request classes to customize documentation of your API endpoints:
```

File: readme.md [hunk 18]
```diff
@@ -107,7 +128,7 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
  - **@description** - Implementation Notes
  - **@_204** - Custom description of response code. You can specify any code as you want.
- - **@some_field** - Description of the field from the rules method
+ - **@first_name** - Description of the field from the rules method
```
Comment (astorozhevsky) [inline readme.md:131]: please revert

File: readme.md [hunk 19]
```diff
@@ -50,13 +50,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+        $response = $this->json('put', '/users/1', $data);
 
+        $response->assertStatus(Response::HTTP_NO_CONTENT);
+    }
+ ```
+3. Create request file code.
+
+For correct working of plugin you have to dispose all the validation rules in the rules() method of `YourRequest` class,
+which must be connected to the controller via DependencyInjection. In annotation of custom request you can specify
+summary and description. Plugin will take validation rules from your request and use it as description
```
Comment (astorozhevsky) [inline readme.md:79]:
```suggestion
Plugin will take validation rules from your request and use it as description
```

File: readme.md [hunk 20]
```diff
@@ -107,7 +128,7 @@ plugin is able to draw Swagger-template to display the generated documentation f
  - **@summary** - short description of request
  - **@description** - Implementation Notes
  - **@_204** - Custom description of response code. You can specify any code as you want.
- - **@some_field** - Description of the field from the rules method
+ - **@first_name** - Description of the field from the rules method
  
  If you do not create a class Request, the summary, Implementation Notes and parameters will be empty.
```
Comment (astorozhevsky) [inline readme.md:133]: please remove the text below to the line `so, you can specify way to collect documentation by creating your custom data collector class.`

File: readme.md [hunk 21]
```diff
@@ -124,7 +145,12 @@ plugin is able to draw Swagger-template to display the generated documentation f
  - *auto-doc.route* - it's a route for generated documentation  
  - *auto-doc.basePath* - it's a root of your api root
  
-Also you can specify way to collect documentation by creating your custom data collector class.
+Also, you can specify way to collect documentation by creating your custom data collector class.
+
+### Test result
```
Comment (astorozhevsky) [inline readme.md:150]: please remove the heading

File: readme.md [hunk 22]
```diff
@@ -124,7 +145,12 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+
+![img.png](assets/images/img.png)
```
Comment (astorozhevsky) [inline readme.md:152]: add the steps before:

4. Run tests
5. Go to route defined in config `auto-doc.route`
6. Profit!

And paster the image here.

**Review by astorozhevsky — CHANGES_REQUESTED** [review]

File: readme.md [hunk 23]
```diff
@@ -49,14 +49,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
-## Usages
- For correct working of plugin you have to dispose all the validation rules in the rules() method of `YourRequest` class, 
- which must be connected to the controller via DependencyInjection. In annotation of custom request you can specify 
- summary and description. Plugin will take validation rules from your request and use it as description 
- of input parameter. 
-  
-### Example
+## Usage
 
+1. Create request data fixture
```
Comment (DenTray) [inline readme.md:54]: let's remove the first point and use permanent data inside the test case

File: readme.md [hunk 24]
```diff
@@ -49,14 +49,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+public function testUpdate()
+{
+    $data = json_decode(file_get_contents('update_user.json'), true);
+
+    $response = $this->json('put', '/users/1', $data);
+
+    $response->assertStatus(Response::HTTP_NO_CONTENT);
+}
+ ```
+3. Create request file code.
```
Comment (DenTray) [inline readme.md:73]:
```suggestion
3. Create request class.
```

File: readme.md [hunk 25]
```diff
@@ -49,14 +49,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+
+For correct working of plugin you have to dispose all the validation rules
```
Comment (DenTray) [inline readme.md:75]: please place it under the class example

File: readme.md [hunk 26]
```diff
@@ -49,14 +49,34 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
   - Call `php artisan swagger:push-documentation` console command after the `tests` stage in your CI/CD configuration
 
-## Usages
- For correct working of plugin you have to dispose all the validation rules in the rules() method of `YourRequest` class, 
- which must be connected to the controller via DependencyInjection. In annotation of custom request you can specify 
- summary and description. Plugin will take validation rules from your request and use it as description 
- of input parameter. 
-  
-### Example
+## Usage
```
Comment (DenTray) [inline readme.md:52]: let's extend the structure

```php
## Usage

### Basic usage
...
### Annotations
...
### Configs
...

### Custom driver
```

File: readme.md [hunk 27]
```diff
@@ -95,36 +115,39 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
- 1. Annotations of request
- 2. Default description from *auto-doc.defaults.code-descriptions.{$code}*
- 3. Descriptions from **Symfony\Component\HttpFoundation\Response::$statusTexts**
-  
-  Note about configs:  
- - *auto-doc.route* - it's a route for generated documentation  
- - *auto-doc.basePath* - it's a root of your api root
- 
-Also you can specify way to collect documentation by creating your custom data collector class.
+If you do not create a class Request, the summary, Implementation Notes and parameters will be empty.
```
Comment (DenTray) [inline readme.md:133]: let's mark this sentence as a note

File: readme.md [hunk 28]
```diff
@@ -95,36 +115,39 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+So, you can specify the way to collect documentation by creating your custom data collector class.
```
Comment (DenTray) [inline readme.md:134]: Please remove this sentece and describe the "Custom driver" block

File: readme.md [hunk 29]
```diff
@@ -95,36 +115,39 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+
+4. Create Controller file code
```
Comment (DenTray) [inline readme.md:136]:
```suggestion
4. Inject the request class into controller method
```

File: readme.md [hunk 30]
```diff
@@ -95,36 +115,39 @@ plugin is able to draw Swagger-template to display the generated documentation f
...
+```php
+public function update(UpdateUserDataRequest $request, $id)
+{
+    User::where('id', $id)->update($request->validated());
+    
+    return response('', Response::HTTP_NO_CONTENT);
+}
+```
+5. Run tests
+6. Go to route defined in config auto-doc.route
```
Comment (DenTray) [inline readme.md:146]:
```suggestion
6. Go to route defined in the `auto-doc.route` config
```

**Review by astorozhevsky — APPROVED** [review]

File: readme.md [hunk 31]
```diff
@@ -1,130 +1,158 @@
...
 </p>
 
 ## Introduction
-This plugin is designed to generate documentation about your Rest API while 
-passing the tests. Special Middleware installed on the Route generates 
-Swagger-file after the successful completion of all tests. In addition, this 
-plugin is able to draw Swagger-template to display the generated documentation for a config.
+
+This plugin is designed to generate documentation for your REST API during the 
+passing the tests.
```
Comment (DenTray) [inline readme.md:14]:
```suggestion
passing PHPUnit tests.
```

File: readme.md [hunk 32]
```diff
@@ -1,130 +1,158 @@
...
-     public function authorize()
-     {
-         return true;
-     }  
+1. Require this package with composer using the following command: `composer require ronasit/laravel-swagger`
+
+    > ***Note***
+    > 
+    > For Laravel 5.5 or later the package will be auto-discovered.
+    > For older versions add the `AutoDocServiceProvider` to the
```
Comment (DenTray) [inline readme.md:23]:
```suggestion
    > For elder versions add the `AutoDocServiceProvider` to the
```

File: readme.md [hunk 33]
```diff
@@ -1,130 +1,158 @@
...
+    > 
+    > ```php
+    > 'providers' => [
+    >    // ...
+    >    RonasIT\Support\AutoDoc\AutoDocServiceProvider::class,
+    > ],
+    > ```
+
+ 1. Run `php artisan vendor:publish`
+ 2. Add middleware `\RonasIT\Support\AutoDoc\Http\Middleware\AutoDocMiddleware::class` into `Http/Kernel.php`.
```
Comment (DenTray) [inline readme.md:34]:
```suggestion
 2. Add `\RonasIT\Support\AutoDoc\Http\Middleware\AutoDocMiddleware::class` middleware to the global HTTP middleware stack in `Http/Kernel.php`.
```

File: readme.md [hunk 34]
```diff
@@ -1,130 +1,158 @@
...
+ 3. Add `\RonasIT\Support\AutoDoc\Tests\AutoDocTestCaseTrait` in your `TestCase` in `tests/TestCase.php`
```
Comment (DenTray) [inline readme.md:35]:
```suggestion
 3. Add `\RonasIT\Support\AutoDoc\Tests\AutoDocTestCaseTrait` trait to `tests/TestCase.php`
```

File: readme.md [hunk 35]
```diff
@@ -1,130 +1,158 @@
...
+    * 
+    * @is_active will indicate whether the user is active or not
+    */
+    class UpdateUserDataRequest extends FormRequest
+    {
+        /**
+        * Determine if the user is authorized to make this request.
+        *
+        * @return bool
+        */
```
Comment (DenTray) [inline readme.md:97]:
```suggestion
    {
        /**
         * Determine if the user is authorized to make this request.
         *
         * @return bool
         */
```

File: readme.md [hunk 36]
```diff
@@ -1,130 +1,158 @@
...
+        public function authorize()
+        {
+            return true;
+        }  
+    
+        /**
+        * Validation Rules
+        *
+        * @return array
+        */
```
Comment (DenTray) [inline readme.md:107]:
```suggestion
        /**
         * Validation Rules
         *
         * @return array
         */
```

File: readme.md [hunk 37]
```diff
@@ -1,130 +1,158 @@
...
+        }
+    }
+
+    ```
+
+    > ***Note***
+    > 
+    > For correct working of plugin you'll have to dispose all the validation rules 
+    > in the `rules()` method of your request class. Also, your request class
+    > must be connected to the controller via `dependency injection`.
```
Comment (DenTray) [inline readme.md:124]:
```suggestion
    > must be connected to the controller via [dependency injection](https://laravel.com/docs/9.x/container#introduction).
```

File: readme.md [hunk 38]
```diff
@@ -1,130 +1,158 @@
...
+    > Plugin will take validation rules from your request and use it as description
```
Comment (DenTray) [inline readme.md:125]:
```suggestion
    > Plugin will take validation rules from the request class and generate fields description
```

File: readme.md [hunk 39]
```diff
@@ -1,130 +1,158 @@
...
+> If you do not use request class, the summary and description and parameters will be empty.
+
+### Configs
+
+- `auto-doc.route` - route for generated documentation
+- `auto-doc.basePath` - root of your API
+
+### Custom driver
+
+You can specify way to collect documentation by creating your custom driver.
```
Comment (DenTray) [inline readme.md:154]:
```suggestion
You can specify the way to collect documentation by creating your own custom driver.
```

**Review by DenTray — CHANGES_REQUESTED** [review]

**Review by DenTray — APPROVED** [review]

**Pull request merged**